analyzers/src/SonarAnalyzer.Common/Helpers/TypeHelper.cs

```diff
@@ -47,14 +48,13 @@ internal static class TypeHelper
 
         #region TypeName
 
+        [Obsolete("Replace this with KnownType.Matches")]
```

Comment:
Will be removed in a follow-up PR.

Reply:
It's one more level of indirection to do the same thing but if it helps make the code easier to understand I can also let it. No strong opinion.